anthos-bm-gcp-terraform/main.tf

```diff
@@ -0,0 +1,186 @@
+  controlplane_vm_names      = [for vmName in var.instance_names.controlplane : join("-", [var.hostname_prefix, vmName])]
+  worker_vm_names            = [for vmName in var.instance_names.worker : join("-", [var.hostname_prefix, vmName])]
+  controlplane_vxlan_ips     = [for name in local.controlplane_vm_names : local.vm_vxlan_ip[name]]
+  worker_vxlan_ips           = [for name in local.worker_vm_names : local.vm_vxlan_ip[name]]
+  admin_vm_hostnames         = [for vm in module.admin_vm_hosts.vm_info : vm.hostname]
+  vm_vxlan_ip                = { for idx, vmName in local.vm_names : vmName => "10.200.0.${idx + 2}" }
+  vmHostnameToVmName         = { for vmName in local.vm_names : "${vmName}-001" => vmName }
+  cluster_yaml_file          = "${path.module}/resources/.${var.abm_cluster_id}.yaml"
+  cluster_yaml_template_file = "${path.module}/resources/anthos_gce_cluster.tpl"
+  init_script                = "${path.module}/resources/init.sh"
+  vm_hostnames_str           = join("|", local.vm_hostnames)
+  vm_hostnames = concat(
+    local.admin_vm_hostnames,
+    [for vm in module.controlplane_vm_hosts.vm_info : vm.hostname],
+    [for vm in module.worker_vm_hosts.vm_info : vm.hostname]
+  )
+  vm_internal_ips = join("|", concat(
+    [for vm in module.admin_vm_hosts.vm_info : vm.internalIp],
+    [for vm in module.controlplane_vm_hosts.vm_info : vm.internalIp],
+    [for vm in module.worker_vm_hosts.vm_info : vm.internalIp])
+  )
+  publicIps = merge(
+    { for vm in module.admin_vm_hosts.vm_info : vm.hostname => vm.externalIp },
+    { for vm in module.controlplane_vm_hosts.vm_info : vm.hostname => vm.externalIp },
+    { for vm in module.worker_vm_hosts.vm_info : vm.hostname => vm.externalIp }
+  )
+}
+
+module "enable_google_apis_primary" {
+  source                      = "terraform-google-modules/project-factory/google//modules/project_services"
+  version                     = "10.2.0"
+  project_id                  = var.project_id
+  activate_apis               = var.primary_apis
+  disable_services_on_destroy = false
+}
+
+module "enable_google_apis_secondary" {
+  source  = "terraform-google-modules/project-factory/google//modules/project_services"
+  version = "10.2.0"
+  # fetched from previous module to explicitely express dependency
+  project_id                  = module.enable_google_apis_primary.project_id
+  depends_on                  = [module.enable_google_apis_primary]
+  activate_apis               = var.secondary_apis
+  disable_services_on_destroy = false
+}
+
+module "create_service_accounts" {
+  source  = "terraform-google-modules/service-accounts/google"
+  version = "~> 3.0"
+  # fetched from previous module to explicitely express dependency
+  project_id = module.enable_google_apis_secondary.project_id
+  depends_on = [
+    module.enable_google_apis_primary,
+    module.enable_google_apis_secondary
+  ]
+  names         = ["baremetal-gcr"]
```

Comment:
SA name here is hardcoded in init script, and if someone makes a change here they won't get SA downloaded on admin host. Perhaps add a note here saying change in init script as well if you change SA name.

Reply:
This has been fixed in d398726f78b04da29d47b27e4e83ec3280c9df49